docs/index.rst

```diff
@@ -1,6 +1,6 @@
-####################################
-Qiskit IBM Q Provider documentation
-####################################
+#########################################
+Qiskit IBM Quantum Provider documentation
```

Comment:
IBM Quantum Experience Provider?

Reply:
So we decided on just `IBM Quantum Provider` for now, to match `ibmq-provider` (see PR #542 )